snpdb/models/models_variant.py

```diff
@@ -242,32 +242,109 @@ class AlleleMergeLog(TimeStampedModel):
     message = models.TextField(null=True)
 
 
-@dataclass(frozen=True)
+@dataclass
 class VariantCoordinate(FormerTuple):
+    """ This stores coordinates, when you want to use it, be sure to call either:
+            * as_external_explicit() - if you want to interface with eg VCF
+            * as_internal_symbolic() - to store in internal Database
+    """
     chrom: str
-    pos: int
+    start: int
+    end: int
     ref: str
     alt: str
 
+    def __init__(self, chrom, start, end: int = None, ref: str = None, alt: str = None):
+        self.chrom = chrom
+        self.start = start
+        if ref is None:
+            raise ValueError("ref must be supplied")
+        if alt is None:
+            raise ValueError("alt must be supplied")
+        if end is None:
+            if Sequence.allele_is_symbolic(alt):
+                raise ValueError("Must pass 'end' when using symbolic alt")
+            end = start + abs(len(ref) - len(alt))
+        self.end = end
+        self.ref = ref
+        self.alt = alt
+
     @property
     def as_tuple(self) -> Tuple:
-        return self.chrom, self.pos, self.ref, self.alt
+        return self.chrom, self.start, self.end, self.ref, self.alt
 
     def __str__(self):
-        return f"{self.chrom}:{self.pos} {self.ref}>{self.alt}"
+        if self.is_symbolic():
+            s = f"{self.chrom}:{self.start}-{self.end} {self.ref}>{self.alt}"
+        else:
+            s = f"{self.chrom}:{self.start} {self.ref}>{self.alt}"
+        return s
 
     @staticmethod
     def from_clean_str(clean_str: str):
         if full_match := VARIANT_PATTERN.fullmatch(clean_str):
-            return VariantCoordinate(chrom=full_match.group(1), pos=int(full_match.group(2)), ref=full_match.group(3), alt=full_match.group(4))
+            chrom = full_match.group(1)
+            start = int(full_match.group(2))
+            ref = full_match.group(3)
+            alt = full_match.group(4)
+            end = start + abs(len(ref) - len(alt))
```

Comment:
Here end is always `start + abs(len(ref) - len(alt))` but in the VariantCoordinate constructor, if end is omitted, it's only given a value if not Sequence.allele_is_symbolic(alt).

Should this logic also apply here? (ignore if not)